src/GameLogic/MagicEffectsList.cs

```diff
@@ -93,6 +93,18 @@ public async ValueTask ClearAllEffectsAsync()
         }
     }
 
+    /// <summary>
+    /// Clears the effects after death of the player.
+    /// </summary>
+    public async ValueTask ClearEffectsAfterDeathAsync()
+    {
+        var effectsToRemove = this.ActiveEffects.Values.Where(effect => effect.Definition.StopByDeath).ToList();
+        foreach (var effect in effectsToRemove)
```

Comment:
Is possible to execute the dispose method in Parallel?

Reply:
It makes no sense:
1. The list is probably very small or empty. The overhead would be bigger than the gain.
2. Clearing an effect causes locking, e.g. when sending network packets and updating the magic effects list. So the calls would lock each other anyway.